indra/llcommon/llsd.cpp

```diff
@@ -840,6 +840,23 @@ void LLSD::clear()						{ Impl::assignUndefined(impl); }
 
 LLSD::Type LLSD::type() const			{ return safe(impl).type(); }
 
+bool LLSD::isEmpty() const
+{
+    Type t = type();
+    if (t == TypeUndefined)
+        return true;
+    if (t == TypeBinary)
+        return !asBoolean();
+    if (t == TypeInteger)
+        return !asInteger();
+    if (t == TypeReal)
+        return !asReal();
+    if (t == TypeString || t == TypeMap || t == TypeArray)
+        return !size();
```

Comment:
Shouldn't this check UUIDs and dates? And why not 'switch'?

P.S. Shouldn't this be safe(impl).isEmpty() to warranty all types get covered?